pkg/ui/workspaces/cluster-ui/src/api/clusterSettingsApi.ts

```diff
@@ -20,11 +24,99 @@ export function getClusterSettings(
   req: SettingsRequestMessage,
   timeout: string,
 ): Promise<SettingsResponseMessage> {
+  const urlParams = new URLSearchParams();
+  urlParams.append("unredacted_values", "true");
+  if (req.keys) {
+    urlParams.append("keys", req.keys.join(","));
+  }
+
   return fetchData(
     cockroach.server.serverpb.SettingsResponse,
-    `${ADMIN_API_PREFIX}/settings?unredacted_values=true`,
-    cockroach.server.serverpb.SettingsRequest,
-    req,
+    `${ADMIN_API_PREFIX}/settings?` + urlParams.toString(),
+    null,
+    null,
     timeout,
   );
 }
+
+// Preferred.
+export type GetClusterSettingRequest = {
+  names: string[];
+};
+
+enum ClusterSettingType {
+  BOOLEAN = "b",
+  DURATION = "d",
+  BYTE_SIZE = "z",
+  INTEGER = "i",
+  ENUM = "e",
+  FLOAT = "f,",
+  STRING = "s",
+  VERSION = "m",
+  UNKNOWN = "",
+}
+
+export type ClusterSetting = {
+  name: string;
+  value: string;
+  type: ClusterSettingType;
+  description: string;
+  lastUpdated: moment.Moment | null;
+};
+
+const formatProtoResponse = (
+  data: SettingsResponseMessage,
+): Record<string, ClusterSetting> => {
+  const settingsMap = {} as Record<string, ClusterSetting>;
+
+  const entries = Object.values(data?.key_values);
```

Comment:
I think `Object.values` throws an error if the argument is undefined or null, so having `?.` doesn't really protect from anything. I think we need to do a null check on `data?.key_values` before making the call to `Object.Values`.